lib/semantics/resolve-labels.cc

```diff
@@ -0,0 +1,918 @@
+            parser::MessageFormattedText{
+                "%sCASE statement name mismatch"_err_en_US, selectSubTag});
+      }
+    }
+  }
+
+  // C1033
+  void CheckName(const parser::WhereConstruct &whereConstruct) {
+    const auto &constructName{std::get<std::optional<parser::Name>>(
+        std::get<parser::Statement<parser::WhereConstructStmt>>(
+            whereConstruct.t)
+            .statement.t)};
+    if (!BothEqOrNone(constructName,
+            std::get<parser::Statement<parser::EndWhereStmt>>(whereConstruct.t)
+                .statement.v)) {
+      errorHandler_.Say(currentPosition_,
+          parser::MessageFormattedText{
+              "WHERE construct name mismatch"_err_en_US});
+    }
+    for (const auto &maskedElsewhere :
+        std::get<std::list<parser::WhereConstruct::MaskedElsewhere>>(
+            whereConstruct.t)) {
+      if (!PresentAndEq(
+              std::get<std::optional<parser::Name>>(
+                  std::get<parser::Statement<parser::MaskedElsewhereStmt>>(
+                      maskedElsewhere.t)
+                      .statement.t),
+              constructName)) {
+        errorHandler_.Say(currentPosition_,
+            parser::MessageFormattedText{
+                "ELSEWHERE (<mask>) statement name mismatch"_err_en_US});
+      }
+    }
+    if (std::get<std::optional<parser::WhereConstruct::Elsewhere>>(
+            whereConstruct.t)
+            .has_value()) {
+      if (!PresentAndEq(
+              std::get<parser::Statement<parser::ElsewhereStmt>>(
+                  std::get<std::optional<parser::WhereConstruct::Elsewhere>>(
+                      whereConstruct.t)
+                      ->t)
+                  .statement.v,
+              constructName)) {
+        errorHandler_.Say(currentPosition_,
+            parser::MessageFormattedText{
+                "ELSEWHERE statement name mismatch"_err_en_US});
+      }
+    }
+  }
+
+  // C1134, C1166
+  template<typename A>
+  void CheckLabelContext(const char *const stmtString, const A &constructName) {
+    const auto I{std::find(
+        constructNames_.crbegin(), constructNames_.crend(), constructName)};
+    if (I == constructNames_.crend()) {
+      errorHandler_.Say(currentPosition_,
+          parser::MessageFormattedText{
+              "%s construct-name '%s' is not in scope"_err_en_US, stmtString,
+              constructName.c_str()});
+    }
+  }
+
+  // 6.2.5, paragraph 2
+  void CheckLabelInRange(parser::Label label) {
+    if (label < 1 || label > 99999) {
+      errorHandler_.Say(currentPosition_,
+          parser::MessageFormattedText{
+              "label '%" PRIu64 "' is out of range"_err_en_US, label});
+    }
+  }
+
+  // 6.2.5., paragraph 2
+  void addTargetLabelDefinition(parser::Label label,
+      LabeledStmtClassificationSet labeledStmtClassificationSet) {
+    CheckLabelInRange(label);
+    const auto pair{
+        programUnits_.back().targetStmts.emplace(std::make_pair(label,
```

Comment:
I don't think make_pair is needed with emplace.

Reply:
? ... ok, fixed

Reply:
The arguments to `emplace` (and related STL member functions) are forwarded to the constructor of the new element.  If one uses `emplace` with a already-constructed temporary object, the new element's move constructor will be used, assuming that it has one.  The whole point of `emplace` is that it omits the use of an intermediate temporary object.  The runtime savings is probably not going to affect the compiler's performance significantly, but `emplace` should be used anyway when constructing a new object in a container that provides the interface.

Reply:
right. The make_pair was a transient edit that was finally removed but mistakenly pushed in the midst of revisions.